src/ImageRunModal.jsx

```diff
@@ -393,17 +419,43 @@ export class ImageRunModal extends React.Component {
 
                         for (const result of reply) {
                             if (result.status === "fulfilled") {
-                                imageResults = imageResults.concat(JSON.parse(result.value));
+                                try {
+                                    imageResults = imageResults.concat(JSON.parse(result.value));
+                                } catch (err) {
+                                    console.error(err);
+                                }
                             } else {
+                                // manifests query returns 404 Not Found in case used URL is not an image
+                                // supress this error as it will either result in "No images found" or prioritize other erros from search queries
+                                if (result.reason.status === 404) {
+                                    continue;
+                                }
+                                // Supress "could't search registry" error when local image is found
+                                if (result.reason.message.includes("couldn't search registry") && imageExistsLocally(value, this.props.localImages)) {
+                                    continue;
+                                }
```

Comment:
I get it that an image which is locally present needs some special treatment. But the main use case here is that someone copy&pastes a full image name like "ghcr.io/cockpit-project/tasks" into the input box and expects our page to download and run that exact image. I suppose that triggers the "couldn't search registry" case (as ghcr.io doesn't have that  search API), but it would't be present locally already. How/where is that handled?

Reply:
In the case where you'd start typing to search for `ghcr.io/cockpit/ta`. This will start two searches. The standard search query and the manifests query. Standard search will return status 500 and the manifest search will return 404 status.
This is here to cover the standard search which returns 500 but only supresses the warning when regex in `imageExistsLocally` can match an image.

> but it would't be present locally already. How/where is that handled?

the case where manifests search is successful, image does not exist locally and other search queries fail is handled in the code bellow (also the following comment under this)